Crypto (#687)
Gp2mv3/Syntheses: src/q7/crypto-MAT2450/exercises/crypto-MAT2450-exercises.tex

\newcommand{\xor}{\oplus}
\newcommand{\st}{\text{ s.t. }}
\renewcommand{\Z}{\mathbb{Z}}
\DeclareMathOperator{\PrivK}{PrivK}
\DeclareMathOperator{\MacForge}{MacForge}
\DeclareMathOperator{\Sigforge}{Sig-forge}
\DeclareMathOperator{\Invert}{Invert}
\DeclareMathOperator{\DDH}{DDH}
\DeclareMathOperator{\DLog}{DLog}
\newcommand{\PrivKmultcpa}{\PrivK^{\text{multcpa}}}
\newcommand{\Sigforgeone}{\Sigforge^{\text{1-time}}}

% New commands
\newcommand{\flec}{\draw[->,>=latex]}
\newcommand{\flect}{\draw[->,>=latex,red]}
\newcommand{\flecc}{\draw[->,>=latex,blue]}
\newcommand{\mess}{node[midway,above]}
\newcommand{\messd}{node[midway,below]}
\newcommand{\trapeze}[2]{\draw #2 node {#1};
	\draw[line width=1pt] #2 ++(-1,2) -- ++(2,-1) {[rounded corners] -- ++(0,-2) -- ++(-2,-0)} -- ++(0,3);}
\newcommand{\position}{(0,0)}
\newcommand{\mylabel}{$x_1$}
\newcommand{\carref}[2] { \draw[line width = 1pt,fill=#2] #1 ++ (-1,-1) rectangle #1 ++ (1,1); }
\newcommand{\carrep}[3] { \draw[line width = 1pt,pattern color=#2,pattern= #3] #1 ++ (-1,-1) rectangle #1 ++ (1,1); }

\titleformat
{\subsection}[hang]{\bfseries\Large}{\thesubsection: }{0.5ex}{}

\hypertitle{Cryptography}{7}{MAT}{2450}

> {Benoît Legat, Luis Tascon Gutierrez, Guillaume Gheysen, Olivier Leblanc}

Reviewer: ```suggestion
{Benoît Legat \and Luis Tascon Gutierrez \and Guillaume Gheysen \and Olivier Leblanc}
```

> {François Koeune, Olivier Pereira}

Reviewer: ```suggestion
{François Koeune and Olivier Pereira}
```

\newpage

> Special thanks to  Francesco Berti (francesco.berti@uclouvain.be) and Pierrick M\'eaux (pierrick.meaux @uclouvain.be), the assistants of this course who gave us the \LaTeX  code of the statements of the exercises.

Reviewer 2: ```suggestion
Special thanks to  Francesco Berti (francesco.berti@uclouvain.be) and Pierrick M\'eaux (pierrick.meaux@uclouvain.be), the assistants of this course who gave us the \LaTeX  code of the statements of the exercises.
```

> Note: I (Luis Tascon Gutierrez) had to do a merge between the \LaTeX code of the solution we have written and the code the assistants sended to me and it means that there might still be some typo error due to command that were not the same between the two documents.

Reviewer 2: ```suggestion
Note: I (Luis Tascon Gutierrez) had to do a merge between the \LaTeX code of the solution we have written and the code the assistants sent to me and it means that there might still be some typo errors due to commands that were not the same between the two documents.
```

\newcommand{\xor}{\oplus}
\newcommand{\st}{\text{ s.t. }}
\renewcommand{\Z}{\mathbb{Z}}
\DeclareMathOperator{\PrivK}{PrivK}
\DeclareMathOperator{\MacForge}{MacForge}
\DeclareMathOperator{\Sigforge}{Sig-forge}
\DeclareMathOperator{\Invert}{Invert}
\DeclareMathOperator{\DDH}{DDH}
\DeclareMathOperator{\DLog}{DLog}
\newcommand{\PrivKmultcpa}{\PrivK^{\text{multcpa}}}
\newcommand{\Sigforgeone}{\Sigforge^{\text{1-time}}}

% New commands
\newcommand{\flec}{\draw[->,>=latex]}
\newcommand{\flect}{\draw[->,>=latex,red]}
\newcommand{\flecc}{\draw[->,>=latex,blue]}
\newcommand{\mess}{node[midway,above]}
\newcommand{\messd}{node[midway,below]}
\newcommand{\trapeze}[2]{\draw #2 node {#1};
	\draw[line width=1pt] #2 ++(-1,2) -- ++(2,-1) {[rounded corners] -- ++(0,-2) -- ++(-2,-0)} -- ++(0,3);}
\newcommand{\position}{(0,0)}
\newcommand{\mylabel}{$x_1$}
\newcommand{\carref}[2] { \draw[line width = 1pt,fill=#2] #1 ++ (-1,-1) rectangle #1 ++ (1,1); }
\newcommand{\carrep}[3] { \draw[line width = 1pt,pattern color=#2,pattern= #3] #1 ++ (-1,-1) rectangle #1 ++ (1,1); }

\titleformat
{\subsection}[hang]{\bfseries\Large}{\thesubsection: }{0.5ex}{}

\hypertitle{Cryptography}{7}{MAT}{2450}

> {Benoît Legat and Luis Tascon Gutierrez and Guillaume Gheysen and Olivier Leblanc}

Reviewer 2: Je suis désolé, mais j'avais juste demandé de mettre `and` pour les professeurs, pas les auteurs.
Il faut:
* \and pour les auteurs, car ça crée un espace entre ceux-ci
* and (en EN) ou et (en FR) pour les professeurs car c'est inséré tel quel dans le texte de la 1e page

Aussi, je pense que tu te compliques la tâche en faisant ces petites modifs en local. L'utilité du "suggested changes" que je fais à chaque fois, c'est que tu peux faire ces changements directement sur GitHub en un rien de temps. Donc je te conseille de juste cliquer sur "commit suggestion" ci-dessous, c'est la seule chose à faire. :wink: 
```suggestion
{Benoît Legat \and Luis Tascon Gutierrez \and Guillaume Gheysen \and Olivier Leblanc}
```

Reviewer 3: Je ne suis pas d'accord avec toi Martin : le soucis de ces "suggested changes" c'est que c'est commité sur GitHub mais pas en local, donc si l'auteur a envie de faire par la suite des modifications en local (par exemple faire une grosse modif, réarranger les commits ou autre), il doit pull des changements depuis GitHub, alors que c'est sa propre branche (=awkward).

Reviewer 2: Merci, mais maintenant, git a ajouté des lignes quand tu as pull ta branche (cause de conflit).
Donc il faut que tu supprimes ces lignes (on voit un <<<<<<<<<< HEAD ).

Author: @Jimvy et @MartinBraquet d'autres commentaires ou on est bon?

Reviewer 2: Yes :ok_hand:

> {François Koeune \and Olivier Pereira}

Reviewer 2: Ce n'est pas corrigé il me semble.

> {François Koeune \and Olivier Pereira}

Author: C'est bon @MartinBraquet ?

{\subsection}[hang]{\bfseries\Large}{\thesubsection: }{0.5ex}{}

\hypertitle{Cryptography}{7}{MAT}{2450}
{Benoît Legat \and Luis Tascon Gutierrez \and Guillaume Gheysen \and Olivier Leblanc}
{François Koeune \and Olivier Pereira}
\newpage
Special thanks to the assistants Francesco Berti (francesco.berti@uclouvain.be) and Pierrick M\'eaux (pierrick.meaux@uclouvain.be) who gave us the \LaTeX\ code of the statements of the exercises.

> Note: I (Luis Tascon Gutierrez) had to merge the \LaTeX\ code of the solution we have written and the code the assistants sent to me and it means that there might still be some typo errors due to commands that were not the same between the two documents.

Author: Oups, c'était pas intentionnel :p Je corrige ça tout de suite

> Note: I (Luis Tascon Gutierrez) had to merge the \LaTeX\ code of the solution we have written and the code the assistants sent to me and it means that there might still be some typo errors due to commands that were not the same between the two documents.

Author: Ah en fait c'est parce que sinon le LaTeX rendu sur le pdf est collé au mot suivant donc, ça donne "LaTeXcode", c'est une solution que j'ai trouvé sur StackOveflow :p

> Note: I (Luis Tascon Gutierrez) had to merge the \LaTeX\ code of the solution we have written and the code the assistants sent to me and it means that there might still be some typo errors due to commands that were not the same between the two documents.

Reviewer 3: A mon avis on pouvait aussi mettre `\LaTeX{}` pour ajouter l'espace manquante
